src/light_data.c

```diff
@@ -588,7 +702,52 @@ void light_initialise(void)
 
 void light_stat_light_map_clear_area(long x1, long y1, long x2, long y2)
 {
-  _DK_light_stat_light_map_clear_area(x1, y1, x2, y2);
+  // _DK_light_stat_light_map_clear_area(x1, y1, x2, y2);
+  unsigned long j,n,x,k,y;
+  unsigned short *p;
+  unsigned short *light_map;
+  k = y1;
+  if ( y2 >= y1 )
+  {
+    y = y1 << 8;
+    unsigned long i = x1 + (y1 << 8);
+    struct Map *Mapblk1 = get_map_block_at_pos(i);
+    light_map = &game.lish.stat_light_map[i];
+    do
+    {
+      j = x1;
+      if ( x2 >= x1 )
+      {
+        p = light_map;
+        n = k - 1;
+        if ( k - 1 <= 0 )
+          n = 0;
+        struct Map *Mapblk2 = get_map_block_at_pos((n << 8) + x1);
+        do
+        {
+          x = j - 1;
+          if ( x <= 0 )
+            x = 0;
+          struct Map *Mapblk3 = get_map_block_at_pos(x + y);
+          struct Map *Mapblk4 = get_map_block_at_pos((n << 8) + x);
+          if ( Mapblk1->data & 0x7FF && Mapblk2->data & 0x7FF && Mapblk3->data & 0x7FF && Mapblk4->data & 0x7FF )
```

Comment:
0x7FF is some `struct Map`-related flag (not is solid?)

Reply:
It looks to me to be some sort of column ID.

Reply:
I've no idea what it is, though, and evidently neither does anyone else....

Reply:
You may search for `->data & 0x7FF` expression and you could find that it is used as column_idx. 
So best way would be to create some function 
```
unsigned short get_column_idx(struct Map* blk)
{
return blk->data & 0x7FF;
}
```
use it here and in other places too